File: src/main/java/au/id/micolous/metrodroid/transit/troika/TroikaUnknownBlock.java
package au.id.micolous.metrodroid.transit.troika;

import java.util.ArrayList;
import java.util.List;

import au.id.micolous.farebot.R;
import au.id.micolous.metrodroid.ui.HeaderListItem;
import au.id.micolous.metrodroid.ui.ListItem;

class TroikaUnknownBlock extends TroikaBlock {
    TroikaUnknownBlock(byte[] rawData) {
        super(rawData);
    }

    @Override
    public List<ListItem> getInfo() {
        List<ListItem> items = new ArrayList<>();
        items.add(new HeaderListItem(getHeader(mTicketType)));
        items.add(new ListItem(R.string.troika_layout, Integer.toHexString(mLayout)));
        return items;
    }
}

File: src/commonMain/kotlin/au/id/micolous/metrodroid/transit/troika/TroikaUnknownBlock.kt
package au.id.micolous.metrodroid.transit.troika

import au.id.micolous.metrodroid.multi.Parcelize
import au.id.micolous.metrodroid.multi.R
import au.id.micolous.metrodroid.ui.HeaderListItem
import au.id.micolous.metrodroid.ui.ListItem
import au.id.micolous.metrodroid.util.ImmutableByteArray

@Suppress("CanBeParameter")
@Parcelize
internal class TroikaUnknownBlock(private val rawData: ImmutableByteArray) : TroikaBlock(rawData) {
    override val info: List<ListItem>?
        get() = listOf(HeaderListItem(getHeader(mTicketType)),
                ListItem(R.string.troika_layout, mLayout.toString(16)))
}
